libs/cart/state/src/effects/cart-resolver.effects.ts

```diff
@@ -60,29 +64,32 @@ export class DaffCartResolverEffects<T extends DaffCart = DaffCart>
 				),
 				switchMap(({ id }) => this.driver.get(id)),
 				map(resp => new DaffResolveCartSuccess(resp)),
-				catchError(error => {
+				catchError((error: Error) => {
 					switch (true) {
 						case error instanceof DaffServerSideStorageError:
-							return of(new DaffResolveCartServerSide());
+							return of(new DaffResolveCartServerSide(this.errorMatcher(
+                new DaffCartServerSideResolutionError(error.message)
+              )));
 						case error instanceof DaffStorageServiceError:
-							error.message =
-								'Cart resolution failed while attempting to retrieve the cart ID from storage.';
-							return of(new DaffResolveCartFailure(this.errorMatcher(error)));
+							return of(new DaffResolveCartFailure(this.errorMatcher(
+                new DaffCartStorageResolutionError(error.message)
+              )));
 						case error instanceof DaffCartNotFoundError:
 							return this.driver.create().pipe(
 								switchMap(({ id }) => this.driver.get(id)),
 								map(resp => new DaffResolveCartSuccess(resp)),
 								catchError((innerError: DaffError) => {
-									innerError.message =
-										'Cart resolution failed after attempting to generate and load a new cart.';
 									return of(
-										new DaffResolveCartFailure(this.errorMatcher(innerError)),
+										new DaffResolveCartFailure(this.errorMatcher(
+                      new DaffCartNotFoundOrCreatedResolutionError(innerError.message)
+                    )),
 									);
 								}),
 							);
 						default:
-							error.message = 'Cart resolution has failed.';
-							return of(new DaffResolveCartFailure(this.errorMatcher(error)));
+							return of(new DaffResolveCartFailure(this.errorMatcher(
```

Comment:
Just to make sure I understand the errorMatcher function...its purpose is to give the app dev a way to transform errors into state differently if they want to? So if the state that they define has some extra field in addition to `code` and `message` that they want to map to as well

Reply:
Actually I don't think typescript will allow additional fields to be defined on the `DaffStateError` object. The main use case is either changing the error message or splitting a general error into more specific errors based on the message.

Reply:
> The main use case is either changing the error message or splitting a general error into more specific errors based on the message.

Ok, so it's just for allowing the app dev to do this? Because right now, we just map `code => code` and `message => message` by default, right?

Reply:
> Ok, so it's just for allowing the app dev to do this? Because right now, we just map code => code and message => message by default, right?

Correct